articles/quickstart/native/android-vnext/00-login.md

```diff
@@ -0,0 +1,248 @@
+                        "Failure: <%= "${exception.code}" %>",
+                        Snackbar.LENGTH_LONG
+                    ).show()
+                }
+            }
+
+            // Called when authentication completed successfully
+            override fun onSuccess(credentials: Credentials) {
+                runOnUiThread {
+                    Snackbar.make(
+                        binding.root,
+                        "Success: <%= "${credentials.idToken}" %>",
+                        Snackbar.LENGTH_LONG
+                    ).show()
+                }
+            }
+        })
+}
+```
+
+After you call the `WebAuthProvider#start` function, the browser launches and shows the login page. Once the user authenticates, the callback URL is called. The callback URL contains the final result of the authentication process.
+
+:::note
+There are many options to customize the authentication with the `WebAuthProvider` builder. You can read about them in the [Auth0 SDK for Android documentation](/libraries/auth0-android).
+
+<div class="phone-mockup">
+  <img src="/media/articles/native-platforms/android/login-android.png" alt="Mobile example screenshot" />
+</div>
+:::
+
+:::panel Checkpoint
+Add a button to your application that calls `loginWithBrowser`. When you click it, verify that your Android application redirects you to the [Auth0 Universal Login](https://auth0.com/universal-login) page and that you can now log in or sign up using a username and password or a social provider.
+
+Once that's complete, verify that Auth0 redirects back to your app.
+:::
+
+## Add Logout to your App
+
+Use WebAuthProvider to remove the cookie set by the Browser at authentication time, so that the users are forced to re-enter their credentials the next time they try to authenticate.
+
+Add a `logout` method to your app to remove the user's session and log them out of the app:
+
+```kotlin
+private fun logout() {
+  WebAuthProvider.logout(account)
+    .withScheme("demo")
+    .start(this, object: VoidCallback {
+      override fun onSuccess(payload: Void?) {
+      }
+
+      override fun onFailure(error: Auth0Exception) {
+        runOnUiThread {
+          Snackbar.make(
+                  binding.root,
+                  "Failure: <%= "${error.message}" %>",
+                  Snackbar.LENGTH_LONG
+          ).show()
+        }
+      }
+    })
+}
+```
+
+The logout is achieved by using the WebAuthProvider class. This call will open the Browser and navigate the user to the logout endpoint. If the log out is cancelled, you might want to take the user back to where they were before attempting to log out.
+
+:::panel Checkpoint
+Add a button to your app that calls `logout` and logs the user out of your application. When you click it, verify that your Android app redirects you logout page and back again, and that you are no longer logged in to your application.
+:::
+
+## Show User Profile Information
+
+[placeholder]
```

Comment:
This is to come once the user profile changes have been made and we can see what this code looks like.

Reply:
Not sure what we were doing before, but we could pull the user profile using the /userinfo endpoint. See

```kotlin
public fun userInfo(accessToken: String): Request<UserProfile, AuthenticationException>
```